### ISSUE TITLE
Unqualified threshold checks contradict logits support

### ISSUE THREAD

**Issue by MichaelSpece** [PyTorchLightning/metrics#369]
## 🐛 Bug

Accuracy both functional and module-wise are currently designed to accommodate logits.  However, the most natural choice of a threshold (`0`) fails in both cases.  More generally, thresholds outside of (0, 1) trigger `ValueError`.  This is documented behavior that contradicts logits support.

#### Code sample

`torchmetrics.functional.classification.accuracy(torch.tensor([0]), torch.tensor([1]), threshold=0)`

or

`torchmetrics.classification.accuracy.Accuracy(threshold=0)`

### Expected behavior

Ability to choose arbitrary float thresholds without triggering an exception.

### Environment

Bugs are apparent from the source code itself.  The threshold checks are unqualified.

### Additional context

Predictions are formed from a given `preds` via `preds >= threshold`.

Threshold checks were recently removed for F1 (#350, #351) to support logits.

Open to assignment.

**Comment by SkafteNicki** [comment]
Fixed in PR https://github.com/PyTorchLightning/metrics/pull/351, please update to master:
```
pip install git+https://github.com/PytorchLightning/metrics.git@master
```
Closing issue.

**Comment by MichaelSpece** [comment]
Hi @SkafteNicki , Apologies for the confusion, but could we reopen this issue?  I was working off the latest (post-#351) master, as alluded to in my context description.  Please see remaining offending code at the following:
https://github.com/PyTorchLightning/metrics/blob/a5af6d851b4236483a3f34f15ccfa79bb90a728a/torchmetrics/classification/accuracy.py#L142-L143
https://github.com/PyTorchLightning/metrics/blob/a5af6d851b4236483a3f34f15ccfa79bb90a728a/torchmetrics/classification/accuracy.py#L208-L209
https://github.com/PyTorchLightning/metrics/blob/a5af6d851b4236483a3f34f15ccfa79bb90a728a/torchmetrics/functional/classification/accuracy.py#L244-L245
https://github.com/PyTorchLightning/metrics/blob/a5af6d851b4236483a3f34f15ccfa79bb90a728a/torchmetrics/functional/classification/accuracy.py#L274-L275
Logits support (with thresholds) is claimed here: 
https://github.com/PyTorchLightning/metrics/blob/a5af6d851b4236483a3f34f15ccfa79bb90a728a/torchmetrics/classification/accuracy.py#L56
https://github.com/PyTorchLightning/metrics/blob/a5af6d851b4236483a3f34f15ccfa79bb90a728a/torchmetrics/functional/classification/accuracy.py#L210

**Comment by SkafteNicki** [comment]
Hi @MichaelSpece, thanks for making me aware that the code indeed is not updated everywhere.
